### PR TITLE
♻️ Fix Fuzz Edge Case in `setNonce`

### PR DESCRIPTION
**Pull request by pcaversaccio**
### 🕓 Changelog

This PR fixes an edge case in the fuzz tests using the cheat code [`vm.setNonceUnsafe`](https://github.com/foundry-rs/foundry/blob/a7fdd78d4a67ade07e25da8223bc87b22d769609/crates/cheatcodes/spec/src/vm.rs#L340) to prevent the following error ([example](https://github.com/pcaversaccio/createx/actions/runs/6968006231/job/18961091267?pr=35#step:16:5327)):

```bash
[FAIL. Reason: new nonce (0) must be strictly equal to or higher than the account's current nonce (4); counterexample: calldata=0x03b1537b0000000000000000000000001804c8ab1f12e6bbf3894d4083f33e07309d1f380000000000000000000000000000000000000000000000000000000000000000 args=[0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38, 0]] testFuzz_WhenTheNonceValueDoesNotExceed18446744073709551614(address,uint64) (runs: 5443, μ: 2369694, ~: 2369752)
```

I also bump some submodules and outdated dependencies (i.e. `ethers`, `@typescript-eslint/eslint-plugin`, and `@typescript-eslint/parser`) in the `package.json` and `pnpm-lock.yaml` file.

#### 🐶 Cute Animal Picture

image

### CODE REVIEW

File: test/public/CREATE/CreateX.computeCreateAddress_1Arg.t.sol [hunk 1]
```diff
@@ -16,7 +16,7 @@ contract CreateX_ComputeCreateAddress_1Arg_Public_Test is BaseTest {
     /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
 
     modifier whenTheNonceValueDoesNotExceed18446744073709551614(uint64 nonce) {
-        vm.assume(nonce != 0 && nonce < type(uint64).max);
+        vm.assume(nonce != 0 && nonce < type(uint64).max && nonce >= vm.getNonce(createXAddr));
         vm.setNonce(createXAddr, nonce);
```
Comment (mds1) [inline test/public/CREATE/CreateX.computeCreateAddress_1Arg.t.sol:20]: An alternative to consider is leaving the `assume` statements and switching to the `vm.setNonceUnsafe` cheat which does allow decreasing the nonce, which should be safe since these are tests for view methods. That lets us cover a wider range of nonces

**Review by mds1 — APPROVED** [review]

**Comment by pcaversaccio** [comment]
**Note:** The code changes committed to this PR have been merged as part of https://github.com/pcaversaccio/createx/pull/40.
